test/web/20-session.js

```diff
@@ -0,0 +1,270 @@
+      let keys = Object.keys(session);
+      keys.should.deep.equal(['data', '_service', '_eventTypeListeners']);
+      session.data.should.be.an('object');
+      // an authenticated session has data
+      Object.keys(session.data).should.deep.equal(['account']);
+      session.data.account.should.be.an('object');
+      session.data.account.should.have.property('id');
+      session.data.account.id.should.equal(account.id);
+      await delay(2000);
+      await session.refresh();
+      keys = Object.keys(session);
+      // an unauthenticated session has no data
+      keys.should.deep.equal(['data', '_service', '_eventTypeListeners']);
+      Object.keys(session.data).should.deep.equal([]);
+    });
+    it('should refresh', async function() {
+      let err;
+      try {
+        session = await getSession({id: 'auth-refresh-session', store});
+      } catch(e) {
+        err = e;
+      }
+      should.not.exist(err);
+      should.exist(session);
+      session.should.be.an('object');
+      let keys = Object.keys(session);
+      keys.should.deep.equal(['data', '_service', '_eventTypeListeners']);
+      session.data.should.be.an('object');
+      Object.keys(session.data).should.deep.equal(['account']);
+      session.data.account.should.be.an('object');
+      session.data.account.should.have.property('id');
+      session.data.account.id.should.equal(account.id);
+      for(let i = 0; i < 4; i++) {
+        await delay(250);
+        await session.refresh();
+        keys = Object.keys(session);
+        // an authenticated session has data
+        keys.should.deep.equal(['data', '_service', '_eventTypeListeners']);
+        Object.keys(session.data).should.deep.equal(['account']);
+        session.data.account.should.be.an('object');
+        session.data.account.should.have.property('id');
+        session.data.account.id.should.equal(account.id);
+      }
+    });
+    it('should emit change event on refresh when session expired',
+      async function() {
+        let err;
+        try {
+          session = await getSession({id: 'auth-emit-change', store});
+        } catch(e) {
+          err = e;
+        }
+        should.not.exist(err);
+        should.exist(session);
+        session.should.be.an('object');
+        let keys = Object.keys(session);
+        keys.should.deep.equal(['data', '_service', '_eventTypeListeners']);
+        session.data.should.be.an('object');
+        Object.keys(session.data).should.deep.equal(['account']);
+        session.data.account.should.be.an('object');
+        session.data.account.should.have.property('id');
+        session.data.account.id.should.equal(account.id);
+        const changeEvent = new Promise(resolve => {
+          session.on('change', ({authentication, oldData, newData}) => {
+            should.not.exist(authentication);
+            should.exist(oldData);
+            should.exist(newData);
+            oldData.should.not.deep.equal(newData);
```

Comment:
if any of these things throw (the assertion fails), then it should call `reject(err);` on your Promise.  So, all these assertions should be in a try/catch, catch the error and call `reject` with it.

Reply:
Addressed here: https://github.com/digitalbazaar/bedrock-web-session/pull/9/commits/886ccc59a3e6990576c161773af7940f24466e53